pkg/appctx/container.go

```diff
@@ -57,3 +57,13 @@ func Provide(ctx context.Context, key interface{}, factory func() (interface{},
 
 	return val, nil
 }
+
+func CopyContainer(from, to context.Context) (context.Context, error) {
```

Comment:
Copy implies... well a copy. You are however sharing the container, so if I provide something in one container, it is also available in the other. I guess this is intended, but then the method should be called `ShareContainer`